server/services/v1/database/query_runner.go

```diff
@@ -413,14 +414,22 @@ func (runner *StreamingQueryRunner) buildReaderOptions(collection *schema.Defaul
 		}
 	}
 
+	if config.DefaultConfig.SecondaryIndex.ReadEnabled {
+		if options.iquery, err = runner.buildSecondaryIndexKeysUsingFilter(collection, runner.req.Filter); err == nil {
+			options.secondaryIndex = true
+			return options, nil
+		}
+	}
+
+	// TODO: I think this might need to be changed, we should only support search on the search endpoint
 	if options.filter.None() || !options.filter.IsIndexed() {
```

Comment:
Do we still support search as a reader option or should it only be part of the search API?

Reply:
Created issue https://github.com/tigrisdata/tigris/issues/869 for this